### PR TITLE
feat(acir)!: add `EcdsaSecp256r1` blackbox function

### PR DESCRIPTION
**Pull request by Maddiaa0**
# Description

Adds support for the secp256 r1 curve through a black box function.

Related prs:
acvm-backend-barretenberg: https://github.com/noir-lang/acvm-backend-barretenberg/pull/223
barretenberg: https://github.com/AztecProtocol/barretenberg/pull/576

## Problem\*

Resolves 

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: acvm/src/pwg/blackbox/signature/ecdsa.rs [hunk 1]
```diff
@@ -112,10 +209,11 @@ fn verify_secp256k1_ecdsa_signature(
...
 #[cfg(test)]
 mod test {
-    use super::verify_secp256k1_ecdsa_signature;
+    use super::{verify_secp256k1_ecdsa_signature, verify_secp256r1_ecdsa_signature};
 
     #[test]
-    fn verifies_valid_signature_with_low_s_value() {
+    fn verifies_valid_k1_signature_with_low_s_value() {
+        // let message = "ECDSA proves knowledge of a secret number in the context of a single message";
         // 0x3a73f4123a5cd2121f21cd7e8d358835476949d035d9c2da6806b4633ac8c1e2,
```
Comment (vezenovm) [inline acvm/src/pwg/blackbox/signature/ecdsa.rs:217]: This hash is different than the one below?

File: brillig_vm/src/black_box.rs [hunk 2]
```diff
@@ -219,6 +232,74 @@ fn verify_secp256k1_ecdsa_signature(
...
+    use k256::{
+        elliptic_curve::{
+            sec1::{Coordinates, ToEncodedPoint},
+            IsHigh,
+        },
+        AffinePoint, EncodedPoint, ProjectivePoint, PublicKey,
+    };
+    // Convert the inputs into k256 data structures
+
+    let signature = Signature::try_from(signature.as_slice()).unwrap();
```
Comment (vezenovm) [inline brillig_vm/src/black_box.rs:248]: We can just make an issue to remove code duplication for `verify_secp256k1_ecdsa_signature` and `verify_secp256r1_ecdsa_signature`

Comment (Maddiaa0) [reply brillig_vm/src/black_box.rs:248]: Its commented at the function head

File: acvm/src/pwg/blackbox/signature/ecdsa.rs [hunk 3]
```diff
@@ -151,4 +249,37 @@ mod test {
 
         assert!(valid)
     }
+
+    #[test]
+    fn verifies_valid_r1_signature_with_low_s_value() {
+        // let message = "ECDSA proves knowledge of a secret number in the context of a single message";
+        // 0x54705ba3baafdbdfba8c5f9a70f7a89bee98d906b53e31074da7baecdc0da9ad
```
[the remark this answers is not in the document]
Comment (Maddiaa0) [reply acvm/src/pwg/blackbox/signature/ecdsa.rs:256]: Removing comment, kev and i had a look through git history for the preimage but that clearly isnt it, and it isnt required to ill remove the comments

**Review by TomAFrench — DISMISSED** [review]
Spoke with @kevaundray and he wants to hold this to go out after 0.16.0

**Review by kevaundray — APPROVED** [review]

**Pull request merged**